### PR TITLE
Adopt `@twreporter/redux@7` 

### PR DESCRIPTION
**Pull request by nickhsine**
### Reason to change: 
[Redux refactoring according to go-api v2 doc #120](https://github.com/twreporter/twreporter-npm-packages/pull/120)
[fix(redux): fetchAFullTopic, fetchAFullPost and related-posts-of reducer #1254 ](https://github.com/twreporter/twreporter-npm-packages/pull/125)

### Related PRs:
[refactor(react-components): update listing-page components](https://github.com/twreporter/twreporter-npm-packages/pull/127)
[refactor(index-page): update according to container changed](https://github.com/twreporter/twreporter-npm-packages/pull/126)
[feat(react-article-components): update Relateds props](https://github.com/twreporter/twreporter-npm-packages/pull/124)

### Notable Changes
- update index page
- update article page
- update topic landing page
- update topic list page
- update category list page
- update tag list page
- update testing-server:
  - provide  `/v2/index_page`, `/v2/topics`, `/v2/topics/:slug`, `/v2/posts` and `/v2/posts/:slug` api endpoints.
  - remove `/v1/index_page`, `/v1/index_page_categories`, `/v1/topics`, `/v1/topics/:slug`, `/v1/posts` and `/v1/posts/:slug` api endpoints.

### CODE REVIEW

File: src/mock-data/v2/posts.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+  return rtn
+}
+
+/**
+ *  This function seeks the posts with certain tags or categories.
+ *
+ *  @param {Object[]} mockPosts - array of mocked posts
+ *  @param {string[]} ids - array of ids
+ *  @param {string} listType - category_id or tag_id
```
Comment (nickhsine) [inline src/mock-data/v2/posts.js:41]: 'category_id' -> 'categories'
'tag_id' -> 'tags'

File: src/mock-data/v2/posts.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          rtn.push(post)
+        }
+      })
+    })
+  })
+
+  return rtn
+}
+
+export function mockAPostResponse(slug) {
```
[the remark this answers is not in the document]
Comment (nickhsine) [reply src/mock-data/v2/posts.js]: Please see the latest commits.

File: src/containers/Category.js [hunk 3]
```diff
@@ -120,6 +103,7 @@ class Category extends PureComponent {
           data={posts}
           catName={catLabel}
           isFetching={isFetching}
+          showSpinner={true}
```
[the remark this answers is not in the document]
Comment (nickhsine) [reply src/containers/Category.js:107]: `showSpinner` is used by https://github.com/twreporter/twreporter-npm-packages/blob/master/packages/react-components/src/is-fetching-wrapper.js#L15

**Comment by nickhsine** [comment]
@taylrj 
Please help to review this large PR if you are available.

File: src/mock-data/v2/topics.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import topics from './topics.json'
+import { cloneMetaOfTopic } from './utils'
+
+const mocks = {
+  topics,
+}
+
+export function mockATopicResponse(slug) {
```
[the remark this answers is not in the document]
Comment (nickhsine) [reply src/mock-data/v2/topics.js]: See 1b3e76c , thx

File: src/containers/Topics.js [hunk 5]
```diff
@@ -198,50 +141,241 @@ class Topics extends Component {
...
+  _.forEach(ids, id => {
+    const topic = _.get(topicEntities, id)
+    if (topic) {
+      topics.push(topicProp(topic))
+    }
   })
+
+  const featureTopic = restoreFeatureTopic(state)
+  if (page === firstPage && featureTopic !== null ) {
+    topics[0] = featureTopic
```
[the remark this answers is not in the document]
Comment (nickhsine) [reply src/containers/Topics.js:337]: `featureTopic` is the first topic with three related posts.
`featureTopic` is used in `https://twreporter.org/topics?page=1` and latest topic section of index page.

File: src/containers/Article.js [hunk 6]
```diff
@@ -134,41 +181,41 @@ class Article extends PureComponent {
...
       )
     }
 
-    if (_.get(selectedPost, 'slug') !== _.get(match, 'params.slug')) {
-      return null
+    if (isFetchingPost) {
+      return (
+        <ArticlePlaceholder />
+      )
     }
```
Comment (taylrj) [inline src/containers/Article.js:209]: Render `<SystemError>` when `post` is invalid ? 

```
if (!post) {
      return (
        <SystemError error={{ statusCode: 500 }} />
      )
 }
```

?

**Comment by nickhsine** [comment]
updated.

**Comment by taylrj** [comment]
LGTM 👍 🥇

**Pull request merged**